spatialmath/pose.go

```diff
@@ -165,8 +165,12 @@ func PoseInverse(p Pose) Pose {
 // p1 and p2 are the two poses to interpolate between, by is a float representing the amount to interpolate between them.
 // by == 0 will return p1, by == 1 will return p2, and by == 0.5 will return the pose halfway between them.
 func Interpolate(p1, p2 Pose, by float64) Pose {
+	p2Orient := p2.Orientation().Quaternion()
+	if OrientationBetween(p1.Orientation(), p2.Orientation()).Quaternion().Real < 0 {
```

Comment:
Can reuse p2Orient here

Reply:
I actually can't; take a close look at where the parenthesis are.

What's returned by `.Quaternion()` is a `quat.Number`, which doesn't actually fill the Orientation interface.